src/js/components/accordion.js

```diff
@@ -0,0 +1,270 @@
+
+      switch (event.keyCode) {
+        case keyCodes.up: {
+          if (this.triggers[prevTrigger] === undefined) {
+            this.triggers[this.triggers.length - 1].focus();
+          } else {
+            this.triggers[prevTrigger].focus();
+          }
+          break;
+        }
+
+        case keyCodes.down: {
+          if (this.triggers[nextTrigger] === undefined) {
+            this.triggers[0].focus();
+          } else {
+            this.triggers[nextTrigger].focus();
+          }
+          break;
+        }
+
+        case keyCodes.end: {
+          this.triggers[this.triggers.length - 1].focus();
+          break;
+        }
+
+        case keyCodes.home: {
+          this.triggers[0].focus();
+          break;
+        }
+
+        default: {
+          break;
+        }
+      }
+    }
+  }
+
+  /**
+   * This function is used to initialize accordions on load. It also adds the
+   * appropriate aria-expanded, and hidden attributes to remove some burden
+   * from the develop.
+   */
+  _openOnInit() {
+    // If accordion is set to open all panels on init, open them
+    if (this.element.hasAttribute('data-accordion-init')) {
+      this.panels.forEach((panel) => {
+        this.open(panel);
+      });
+    } else if (this.openOnInit) {
+      // If this.openOnInit has been set, open it on initialization
+      this.open(this.openOnInit);
+      // Keep all other accordions closed by setting their initial states
+      this.panels.forEach((panel, index) => {
+        if (panel !== this.openOnInit) {
+          // Set non-initialized accordion panels to hidden
+          this.panels[index].setAttribute('hidden', '');
+          let trigger = this.element.querySelector(`[${this.triggerAttribute}="${panel.dataset.accordionPanel}"]`);
+          // Set 'aria-expanded' for all non-initialized accordion triggers to 'false'
+          trigger.setAttribute('aria-expanded', 'false');
+        }
+      });
+    } else {
+      // Otherwise, keep all accordions closed to start
+      this.panels.forEach((panel, index) => {
+        // Set all of the accordion panels to hidden
+        this.panels[index].setAttribute('hidden', '');
+        let trigger = this.element.querySelector(`[${this.triggerAttribute}="${panel.dataset.accordionPanel}"]`);
+        // Set 'aria-expanded' for all of the accordion triggers to 'false'
+        trigger.setAttribute('aria-expanded', 'false');
+      });
+    }
+  }
+
+  /**
+  * This function is used to determine whether an allowable number of accordion
+  * panels has been set to open on initialization.
+  * @param {Node} entity - the Node in which the given selector should be found
+  * @param {string} selector - an element selector
+  * @param {Function} callback - a function the user wishes to trigger if the
+  * selector is below the maximum allowed number
+  * @param {number} max - the maximum number of a given selector allowed
+  */
+  _quantifySelector(entity, selector, callback, max = 1) {
```

Comment:
I'm 50/50 on this method/helper. It adds some additional mental overhead. I had to keep jumping back and forth between this definition site and where it's used in the constructor to understand what was happening. If we're doing these kinds of checks in every component, maybe we could refactor it out into a util/helper function and just `import` it where needed? That's said it may be a bit of a premature abstraction when you could just as easily run a check for too many attributes/initialized panels while you're doing all the other set up in the constructor.